packages/typespec-ts/src/modular/serialization/serializers.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  SdkConstantType,
+  SdkContext,
+  SdkEnumType,
+  SdkModelPropertyType,
+  SdkModelType,
+  SdkType,
+  SdkUnionType
+} from "@azure-tools/typespec-client-generator-core";
+import { isDefined } from "@azure/core-util";
+import { getDiscriminator, UsageFlags } from "@typespec/compiler";
+import _ from "lodash";
+import * as Reify from "../../reify/index.js";
+import {
+  getParameterTypePropertyName,
+  getReturnTypePropertyName,
+  SerializerMap,
+  SerializerOutput
+} from "./util.js";
+
+interface SerializeTypeOptions<
+  TCGCType extends SdkType | SdkModelPropertyType
+> {
+  dpgContext: SdkContext;
+  functionType: UsageFlags;
+  serializerMap?: SerializerMap;
+  type: TCGCType;
+  valueExpr: string;
+  importCallback: (importType: ImportType, importedName: string) => void;
+}
+
+const placeholder = <T>(_?: T) => {
+  return '(()=>{throw Error("Not implemented.")})()';
+};
+
+export function serializeType<TCGCType extends SdkType | SdkModelPropertyType>(
+  options: SerializeTypeOptions<TCGCType>
+): SerializerOutput {
+  const { functionType, serializerMap, type, valueExpr } = options;
+  const modularTypeName = (type as { name?: string }).name;
+  const serializerMetadata =
+    isDefined(modularTypeName) && serializerMap?.[modularTypeName];
+
+  if (serializerMetadata) {
+    const functionName =
+      functionType === UsageFlags.Input
+        ? serializerMetadata?.serializerFunctionName
+        : serializerMetadata?.deserializerFunctionName;
+
+    return `${
+      functionName ??
+      (functionType === UsageFlags.Input
+        ? "MISSING_SERIALIZER"
+        : "MISSING_DESERIALIZER")
+    }((${valueExpr}))`;
+  }
+
+  return getSerializeHandler(type.kind)?.(options) ?? valueExpr;
+}
+
+function getSerializeHandler<TCGCType extends SdkType | SdkModelPropertyType>(
+  kind: TCGCType["kind"]
+): ((options: SerializeTypeOptions<TCGCType>) => SerializerOutput) | undefined {
+  type SimpleType =
+    | "any"
+    | "armId"
+    | "azureLocation"
+    | "boolean"
+    | "constant"
+    | "decimal"
+    | "decimal128"
+    | "enum"
+    | "eTag"
+    | "float"
+    | "float32"
+    | "float64"
+    | "guid"
+    | "int16"
+    | "int32"
+    | "int64"
+    | "int8"
+    | "integer"
+    | "ipAddress"
+    | "ipV4Address"
+    | "ipV6Address"
+    | "numeric"
+    | "password"
+    | "safeint"
+    | "string"
+    | "uint16"
+    | "uint32"
+    | "uint64"
+    | "uint8"
+    | "uri"
+    | "url"
+    | "uuid";
+
+  type SerializeHandler<
+    TCGCTypeKind extends (SdkType | SdkModelPropertyType)["kind"]
+  > = {
+    [K in TCGCTypeKind]: (
+      options: SerializeTypeOptions<
+        (SdkType | SdkModelPropertyType) & { kind: K }
+      >
+    ) => SerializerOutput;
+  };
+  type SerializeHandlerMap = Partial<SerializeHandler<SimpleType>> &
+    SerializeHandler<
+      Exclude<(SdkType | SdkModelPropertyType)["kind"], SimpleType>
+    >;
+
+  const handlers: SerializeHandlerMap = {
+    array: placeholder,
```

Comment:
just wonder if we could file issues or have a way to track all those places that we already know it's going to be changed? I notice you added TODO comments in other places.

Reply:
For this in particular, all of the missing functions are contained in this file at the moment. I'll open issues to track remaining work.